lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,17 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Returns a view of this stream that eagerly buffers all events until
+  /// `listen()` is called.
+  Stream<T> buffer() {
+    late StreamSubscription<T> subscription;
```

Comment:
Please don't use `late`. Just make it nullable. Or, preferably, reorder things so it's not necessary.